### DISCUSSION TITLE
Cannot change readonly settings in member

### DISCUSSION THREAD

**Discussion by hamzahshaq** [codefori/vscode-ibmi#1411] [Q&A]
Hi I encounter problem when saving member:
![image](https://github.com/halcyon-tech/vscode-ibmi/assets/112717087/37638741-1602-422a-b879-a183a33cf561)

then I notice that my member have lock icon next to it, where can I change the option?
I already check the filter and connection setting:
![image](https://github.com/halcyon-tech/vscode-ibmi/assets/112717087/e6ebb00e-3916-4371-b92c-4a5773f5fed6)
read only mode not checked

![image](https://github.com/halcyon-tech/vscode-ibmi/assets/112717087/62fb5219-d271-4c74-8c9d-6894109baffe)
protected not checked

already try to change the parameter back and forth (check-unchecked) then reconnect but that doesn't help.

please help.

**Comment by sebjulliand** [comment]
Hi,
can you click on Report an issue and copy and paste the content it generates, so we can get more info about your context.
Also, please share the content of the Code for IBM i output; this may help.
Thanks!

**Reply by hamzahshaq** [reply]
Hi @sebjulliand,

thanks for your respond.

this is text generated from report an issue:
👉🏻 Issue text goes here.

<hr />

⚠️ **REMOVE THIS LINE AND ANY SENSITIVE INFORMATION BELOW!** ⚠️

|Context|Version|
|-|-|
|Code for IBM i version|1.9.4|
|Visual Studio Code version|1.79.2|
|Operating System|win32_x64|


<details>
<summary>Active extensions</summary>

```
Code for IBM i Walkthroughs (vscode-ibmi-walkthroughs): 0.3.1
Emmet (emmet): 1.0.0
Git (git): 1.0.0
Git Base (git-base): 1.0.0
GitHub (github): 0.0.1
GitHub Authentication (github-authentication): 0.0.2
JSON Language Features (json-language-features): 1.0.0
Merge Conflict (merge-conflict): 1.0.0
Microsoft Account (microsoft-authentication): 0.0.1
Node Debug Auto-attach (debug-auto-launch): 1.0.0
TypeScript and JavaScript Language Features (typescript-language-features): 1.0.0
WSL: Recommender (remote-wsl-recommender): 0.0.19
```
</details>


<hr />


<details>
<summary>Remote system</summary>

|Setting|Value|
|-|-|
|IBM i OS|?|
|Tech Refresh|?|
|CCSID|?|
|SQL|Disabled
|Source dates|Disabled

### Enabled features

|/usr/bin|/QSYS.lib/ILEDITOR.lib|/QSYS.LIB|
|-|-|-|
|attr|GETNEWLIBL.PGM|QZDFMDB2.PGM|
|iconv|||
|ls|||
|setccsid|||
|tar|||
</details>



<details>
<summary>Shell env</summary>

```bash
```
</details>



<details>
<summary>Variants</summary>

```json
{
  "american": "#@$",
  "local": "#@$"
}
```
</details>



<details>
<summary>Errors</summary>

```json
[
  {
    "command": "ls -p /QOpenSys/pkgs/bin/",
    "code": 2,
    "stderr": "/QOpenSys/pkgs/bin/ not found",
    "cwd": "/home/hamzah"
  },
  {
    "command": "ls -p /QIBM/ProdData/IBMiDebugService/bin/startDebugService.sh",
    "code": 2,
    "stderr": "/QIBM/ProdData/IBMiDebugService/bin/startDebugService.sh not found",
    "cwd": "/home/hamzah"
  },
  {
    "command": "export BUILDLIB=\"\" && export CURLIB=\"\" && export USERNAME=\"HAMZAH\" && export HOME=\"/home/hamzah\" && export LIBLS=\"\" && env",
    "code": 1,
    "stderr": "bsh: BUILDLIB=: is not an identifier",
    "cwd": "/home/hamzah"
  }
]
```
</details>

and this is output in terminal:
`/home/hamzah: /QOpenSys/usr/bin/qsh
liblist -d  && liblist -c  && liblist -a  && system  "CHKOBJ OBJ(HAMZAHLIB/BULKITA) OBJTYPE(*FILE) AUT(*UPD)"; if [[ $? -ne 0 ]]; then exit 1; fi
{
    "code": 9,
    "signal": null,
    "stdout": "",
    "stderr": ""
}

/home/hamzah: system "DSPFD FILE(HAMZAHLIB/BULKITA) TYPE(*MBR) OUTPUT(*OUTFILE) OUTFILE(ILEDITOR/O_05EtcKYu)"
{
    "code": 0,
    "signal": null,
    "stdout": "CPF9861: Output file O_05ETCKYU created in library ILEDITOR.\nCPF9862: Member O_05ETCKYU added to output file O_05ETCKYU in library ILEDITOR.\nCPF3030: 1 records added to member O_05ETCKYU in file O_05ETCKYU in ILEDITOR.",
    "stderr": ""
}

/home/hamzah: system "QSYS/CPYTOIMPF FROMFILE(ILEDITOR/O_05EtcKYu O_05EtcKYu) TOSTMF('/tmp/vscodetemp-O_RjRVSLSK') MBROPT(*REPLACE) STMFCCSID(1208) RCDDLM(*CRLF) DTAFMT(*DLM) RMVBLANK(*TRAILING) ADDCOLNAM(*SQL) FLDDLM(',') DECPNT(*PERIOD)"
{
    "code": 0,
    "signal": null,
    "stdout": "CPC2958: All records copied from file O_05ETCKYU in ILEDITOR.",
    "stderr": ""
}

.: system "CPYTOSTMF FROMMBR('/QSYS.lib/HAMZAHLIB.lib/BULKITA.file/YEVI01R.mbr') TOSTMF('/tmp/vscodetemp-O_TNEnhTJI') STMFOPT(*REPLACE) STMFCCSID(1208) DBFCCSID(*FILE)"
{
    "code": 0,
    "signal": null,
    "stdout": "CPCA082: Object copied.",
    "stderr": ""
}`

**Reply by hamzahshaq** [reply]
Update on this, I decide to install code for IBM i version 1.9.3 and then the issue fixed. I can save normally. But then I encounter new problem:
When I failed to compile, the problem did not show up on terminal
![image](https://github.com/halcyon-tech/vscode-ibmi/assets/112717087/281067f8-495c-40c8-bfda-3f5368546367)

this is IBM i output:
/home/hamzah: /QOpenSys/usr/bin/qsh
liblist -d  && liblist -c  && liblist -a  && system  "CRTBNDRPG PGM(HAMZAHLIB/YEVI01R) SRCFILE(HAMZAHLIB/BULKITA) SRCMBR(YEVI01R) OPTION(*NODEBUGIO *EVENTF) DBGVIEW(*LIST) TGTRLS(*CURRENT)"; if [[ $? -ne 0 ]]; then exit 1; fi
{
    "code": 9,
    "signal": null,
    "stdout": "",
    "stderr": ""
}

/home/hamzah: system "QSYS/CPYTOIMPF FROMFILE(HAMZAHLIB/EVFEVENT YEVI01R) TOSTMF('/tmp/vscodetemp-O_7tax98Ok') MBROPT(*REPLACE) STMFCCSID(1208) RCDDLM(*CRLF) DTAFMT(*DLM) RMVBLANK(*TRAILING) ADDCOLNAM(*SQL) FLDDLM(',') DECPNT(*PERIOD)"
{
    "code": 255,
    "signal": null,
    "stdout": "",
    "stderr": "CPF2817: Copy command ended because of error."
}


thanks!

**Reply by worksofliam** [reply]
@hamzahshaq What version of IBM i is this?
